YoutubeDownloader/ViewModels/Framework/DialogManager.cs

```diff
@@ -53,31 +57,99 @@ void OnScreenClosed(object? closeSender, EventArgs closeArgs)
         }
     }
 
-    public string? PromptSaveFilePath(string filter = "All files|*.*", string defaultFilePath = "")
+    public async Task<string?> PromptSaveFilePath(string? filter = null, string defaultFilePath = "")
     {
-        var dialog = new VistaSaveFileDialog
+        var topLevel = (Application.Current?.ApplicationLifetime as IClassicDesktopStyleApplicationLifetime)?.MainWindow;
+
+        var storageProvider = topLevel?.StorageProvider;
+        if (storageProvider is null)
         {
-            Filter = filter,
-            AddExtension = true,
-            FileName = defaultFilePath,
-            DefaultExt = Path.GetExtension(defaultFilePath)
-        };
+            return null;
+        }
 
-        return dialog.ShowDialog() == true ? dialog.FileName : null;
+        var filePickResult = await storageProvider.SaveFilePickerAsync(new()
+        {
+            FileTypeChoices = ParseFileTypes(filter),
```

Comment:
We could probably just pass `FileTypeChoices` as an argument to the method, instead of using a string like now